services/search/src/search/routes/filter.py

```diff
@@ -0,0 +1,251 @@
+                    if (
+                        not dataset
+                        or not config
+                        or not split
+                        or not where
+                        or not are_valid_parameters([dataset, config, split, where])
+                    ):
+                        raise MissingRequiredParameterError(
+                            "Parameters 'dataset', 'config', 'split' and 'where' are required"
+                        )
+                    # TODO: validate where
+                    offset = int(request.query_params.get("offset", 0))
+                    if offset < 0:
+                        raise InvalidParameterError(message="Parameter 'offset' must be positive")
+                    length = int(request.query_params.get("length", MAX_NUM_ROWS_PER_PAGE))
+                    if length < 0:
+                        raise InvalidParameterError("Parameter 'length' must be positive")
+                    elif length > MAX_NUM_ROWS_PER_PAGE:
+                        raise InvalidParameterError(
+                            f"Parameter 'length' must not be bigger than {MAX_NUM_ROWS_PER_PAGE}"
+                        )
+                    logger.info(
+                        f'/filter, dataset={dataset}, config={config}, split={split}, where="{where}",'
+                        f" offset={offset}, length={length}"
+                    )
+                with StepProfiler(method="filter_endpoint", step="check authentication"):
+                    # If auth_check fails, it will raise an exception that will be caught below
+                    auth_check(
+                        dataset=dataset,
+                        external_auth_url=external_auth_url,
+                        request=request,
+                        hf_jwt_public_keys=hf_jwt_public_keys,
+                        hf_jwt_algorithm=hf_jwt_algorithm,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                    )
+                # TODO: duplicated in /search
+                with StepProfiler(method="search_endpoint", step="validate indexing was done"):
+                    # no cache data is needed to download the index file
+                    # but will help to validate if indexing was done
+                    duckdb_index_cache_entry = get_cache_entry_from_duckdb_index_job(
+                        processing_graph=processing_graph,
+                        dataset=dataset,
+                        config=config,
+                        split=split,
+                        hf_endpoint=hf_endpoint,
+                        hf_token=hf_token,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                        cache_max_days=cache_max_days,
+                    )
+                    revision = duckdb_index_cache_entry["dataset_git_revision"]
+                    if duckdb_index_cache_entry["http_status"] != HTTPStatus.OK:
```

Comment:
Maybe you will need to add a similar logic as in https://github.com/huggingface/datasets-server/pull/1854/files#diff-b31b94865cf356d2b241fc7f38f98bd9484d61464a991d06364bdf82c2c5ba4eR245 (but it depends which one is going to be merged first).

Reply:
Done.

Reply:
Maybe I am wrong but I saw you added 
`if duckdb_index_cache_entry["content"]["has_fts"] is not True:`
in search.py but I can't see something similar here